backend/src/Permissions.js

```diff
@@ -0,0 +1,49 @@
+const {rule, shield} = require('graphql-shield');
+
+const isAuthenticated = rule({cache: 'contextual'})(
+    async (parent, args, ctx, info) => {
+        const token = ctx.user;
+        if (token) {
+            const user = ctx.dataSources.userDataStore.getUserById(token.uid);
+            if (user) {
+                return true;
+            }
+        }
+        return false;
+    }
+);
+
+const isOwnUser = rule({})(
+    async ({email}, args, ctx, info) => {
+        const token = ctx.user;
+        if (token) {
+            const user = ctx.dataSources.userDataStore.getUserById(token.uid);
+            if (user && user.email === email) {
```

Comment:
cool you did it 🚀 For authorizing read access on User.email
```suggestion
    async (parent, args, ctx, info) => {
        const token = ctx.user;
        if (token) {
            const user = ctx.dataSources.userDataStore.getUserById(token.uid);
            if (user && user.email === parent.email) {
```